src/main.cpp

```diff
@@ -37,62 +38,75 @@ TinyGsmClient client(modem);
 
 #define DEFAULT_BUFFER_SIZE 10
 
-//TinyGsm modem(Serial2);
-//uint32_t rate = 0; // Set to 0 for Auto-Detect
-
 VoltageSensor* voltageSensor;
+Accelerometer* accelSensor;
 
 PollingSensorTask<voltage_t>* voltageSensorTask;
+PollingSensorTask<accel_t>* accelSensorTask;
 
 void setup() {
-   Serial.begin(115200);
-   SerialAT.begin(9600);
-   delay(6000);
+    Serial.begin(115200);
+    SerialAT.begin(9600);
+    delay(6000);
 
-   Serial.println("Starting init");
+    Serial.println("Starting init");
 
-   modem.init();
+    modem.init();
 
-   Serial.println("Initiating GPRS");
+    Serial.println("Initiating GPRS");
 
-   modem.gprsConnect("Fido-core-appl1.apn");
+    modem.gprsConnect("Fido-core-appl1.apn");
 
-   Serial.println("Connected!");
+    Serial.println("Connected!");
+    Serial.printf("Minimal stack size %d\n", configMINIMAL_STACK_SIZE);
+    voltageSensor = new VoltageSensor(DEFAULT_BUFFER_SIZE);
+    accelSensor = new Accelerometer(DEFAULT_BUFFER_SIZE);
+    voltageSensor->addListener([](const voltage_t& newValue) {
+        updateBatteryVoltage(convertVoltageToFloat(newValue));
+    });
 
-   initProperties();
-   Serial.println("Connected to AP");
-   ArduinoCloud.begin(ArduinoIoTPreferredConnection);
+        accelSensor->addListener([](const accel_t& newValue){
+            updateAcceleration(newValue);
+        });
 
+    // })
+    // TODO: Note that voltageSensor will throw an exception if collect is not called before get(). See if we can apply RAII
 
-   Serial.println("Begin Connection");
+    voltageSensorTask = new PollingSensorTask<voltage_t>(voltageSensor, 200, "T_VoltageSensor", 12800 * 100, 5);
+    accelSensorTask = new PollingSensorTask<accel_t>(accelSensor,200,"T_AccelSensor",1280*100,5);
 
-   setDebugMessageLevel(2);
-   ArduinoCloud.printDebugInfo();
+    initProperties();
+    ArduinoCloud.begin(ArduinoIoTPreferredConnection);
 
-   Serial.printf("Minimal stack size %d\n", configMINIMAL_STACK_SIZE);
-   voltageSensor = new VoltageSensor(DEFAULT_BUFFER_SIZE);
+    Serial.println("Begin Connection");
 
-   voltageSensor->addListener([](const voltage_t& newValue) {
-       updateBatteryVoltage(convertVoltageToFloat(newValue));
-   });
+    setDebugMessageLevel(2);
+    ArduinoCloud.printDebugInfo();
 
-   // TODO: Note that voltageSensor will throw an exception if collect is not called before get(). See if we can apply RAII
-
-   voltageSensorTask = new PollingSensorTask<voltage_t>(voltageSensor, 200, "T_VoltageSensor", 1024 * 5, 5);
 }
 
 void loop() {
-   ArduinoCloud.update();
+    Serial.print("Main Loop...\n");
+    vTaskDelay(100);
+    //Serial.printf("Value %d\n", voltageSensor->get());
```

Comment:
We don’t need this block of comments.
The loop() shouldnt have a vTaskDelay according to IoT docs.